content/en/docs/reference/setup-tools/kubeadm/upgrading-v1alpha1-config-to-v1alpha2.md

```diff
@@ -0,0 +1,175 @@
+## Upgrading your `v1alpha1` kubeadm configuration files to `v1alpha2`
+
+This document explains what semantics have changed between `v1alpha1` and `v1alpha2` of the kubeadm config API types, and why. `v1alpha2` was released with kubeadm v1.11.
+
+#### Important notes
+
+- The `v1alpha1` and `v1alpha2` APIs **aren't marked stable** yet, so their semantics will (possibly significantly) change before reaching `v1beta1`. Between `v1beta1` and `v1` the delta is expected to be small/non-existent.
+- **All your `v1alpha1`  configuration files can be read by kubeadm v1.11 and are automatically upgraded to `v1alpha2` internally**. This means you don't _technically_ need to read this document and upgrade each field separately, kubeadm takes care of this automatically under the hood.
+  - The policy is the follwing: If a given API version `v1alphaY` was released in kubeadm v1.Z to replace the existing `v1alphaX` API version, reading `v1alphaX` files in kubeadm v1.Z will work, but kubeadm v1.Z will only _write_ the newer API version `v1alphaY` 
+
+
+#### Using the `kubeadm config migrate` command for automatic conversion
+
+There is  a `kubeadm config migrate` command you might use to convert your old `v1alpha1` configuration files to the newer `v1alpha2` spec. For instance, if you have your existing configuration file `config-v1alpha1.yaml` and want to get the matching `v1alpha2` spec, run the following command:
+
+```
+kubeadm config migrate --old-config config-v1alpha1.yaml --new-config config-v1alpha1.yaml
+```
+
+You can also omit the `--new-config` flag and `kubeadm` will print to `stdout`.
+
+
+### API Changes between `v1alpha1` and `v1alpha2`
+
+#### The `.CloudProvider` and `.PrivilegedPods`fields are gone
+
+In-tree cloud providers that `.CloudProvider` once enabled on a best-effort, experimental 
+basis are deprecated, and use of [out-of-tree cloud providers](#TODO) is encouraged instead.
```

Comment:
Link is missing here

Reply:
> The experimental usage of in-tree cloud providers with the `.CloudProvider` field is deprecated. Usage of [out-of-tree cloud providers](#TODO) is encouraged instead.

Reply:
not explicitly mentioned in any public docs... in-tree has a [deprecation warning][warning] but does not support the assertion that out-of-tree is the thing to use and the [cloud controller manager][ccm] really doesn't really say that external is encouraged over in-tree, it just tells you how to use external provider running in-tree code.

I wonder if this decision was explicitly stated and documented somewhere. The opening paragraphs [in this document][cloud] almost get at it... 

[cloud]: https://docs.google.com/document/d/1YqsPzeG10E59vea5StKFM6W_xZq034eTaf1n649JQvs/edit#

[warning]: https://github.com/kubernetes/kubernetes/tree/master/pkg/cloudprovider#deprecation-notice-this-directory-has-entered-maintenance-mode-and-will-not-be-accepting-new-providers-cloud-providers-in-this-directory-will-continue-to-be-actively-developed-or-maintained-and-supported-at-their-current-level-of-support-as-a-longer-term-solution-evolves
[ccm]: https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/